application/event_processor/event_processor.py

```diff
@@ -121,8 +122,8 @@ def lambda_handler(event: Dict[str, Any], context: LambdaContext) -> None:
     for env_var in EXPECTED_ENVIRONMENT_VARIABLES:
         if env_var not in environ:
             logger.error(f"Environmental variable {env_var} not present")
-
-    change_event = extract_event(event)
+    message = next(event.records).body
```

Comment:
There is an assumption here that there is only 1 record which is how we have it configured and how we expect it to work, we should therefore throw an exception if there is more than 1 record (i.e. if someone changes the configuration) because if that were to happen the x-1 messages would be removed from the queue but not actually processed and therefore lost

Reply:
Please can you check if the implementation is good enough

Reply:
assuming the for loop doesnt exhaust the iterator such that if their is only one item then `next` still returns the item